packages/admin-ui/src/number-input/number-input.tsx

```diff
@@ -0,0 +1,137 @@
+import type { ReactNode, ComponentPropsWithoutRef, Ref } from 'react'
+import React, { useRef, forwardRef } from 'react'
+import { useForkRef, useId } from '@vtex/admin-ui-hooks'
+
+import { IconCaretDown, IconCaretUp } from '@vtex/phosphor-icons'
+import { Button } from '../button'
+import { TextInputContainer as InputContainer } from '../text-input/text-input-container'
+import { TextInputElement as InputElement } from '../text-input/text-input-element'
+import { TextInputTerm as InputTerm } from '../text-input/text-input-term'
+import {
+  FormControl,
+  FormControlLabel,
+  FormControlMessage,
+} from '../form-control'
+
+import type { NumberInputValue } from './number-input.state'
+import { useNumberInput } from './number-input.state'
+import { useMessageFormatter } from '../i18n'
+
+import { messages } from '../form-control/form-control.i18n'
+
+import * as style from './number-input.style'
+
+export const NumberInput = forwardRef(
+  (props: NumberInputProps, ref: Ref<HTMLInputElement>) => {
+    const {
+      prefix,
+      suffix,
+      disabled,
+      optional,
+      error,
+      errorText,
+      helpText,
+      label,
+      id: defaultId,
+      value = '',
+      step = 1,
+      min = Number.NEGATIVE_INFINITY,
+      max = Number.POSITIVE_INFINITY,
+      onChange = () => {},
+      placeholder,
+      ...inputProps
+    } = props
+
+    const { getInputProps, getDecrementButtonProps, getIncrementButtonProps } =
+      useNumberInput({
+        initialValue: value,
+        step,
+        min,
+        max,
+        onChange,
+      })
+
+    const id = useId(defaultId)
+
+    const formatMessage = useMessageFormatter(messages.formControl)
+    const optionalPlaceholder =
+      !label && optional ? formatMessage('optional') : ''
+
+    const inputRef = useRef<HTMLInputElement>(null)
+    const handleClick: React.MouseEventHandler<HTMLDivElement> = () => {
+      if (inputRef.current) {
+        inputRef.current.focus()
+      }
+    }
+
+    // Avoid losing input focus when the spin button is disabled
+    const handleMouseDown: React.MouseEventHandler<HTMLDivElement> = (
+      event
+    ) => {
+      event.preventDefault()
+    }
+
+    return (
+      <FormControl error={error} optional={optional}>
+        {label && <FormControlLabel htmlFor={id}>{label}</FormControlLabel>}
+        <InputContainer
+          onClick={handleClick}
```

Comment:
I think this makes the button lose its focus when clicking using the keyboard
![2022-05-24 16-27-07 2022-05-24 16_27_31](https://user-images.githubusercontent.com/20579226/170116648-8bc07c0b-4b5d-4d16-98b9-57025bbc602a.gif)

Reply:
I am still thinking he best way to solve this, because it is important to keep the focus on the component when we click the button.